blosc/CMakeLists.txt

```diff
@@ -172,11 +172,17 @@ if(COMPILER_SUPPORT_SSE2)
             set_source_files_properties(
                     shuffle-sse2.c bitshuffle-sse2.c blosclz.c fastcopy.c
                     PROPERTIES COMPILE_FLAGS "/arch:SSE2")
+            set_property(
+                    SOURCE shuffle.c
+                    APPEND PROPERTY COMPILE_OPTIONS "/arch:SSE2")
```

Comment:
Using [`COMPILE_OPTIONS`](https://cmake.org/cmake/help/latest/prop_tgt/COMPILE_OPTIONS.html) here since [`COMPILE_FLAGS`](https://cmake.org/cmake/help/latest/prop_tgt/COMPILE_FLAGS.html) is deprecated (see https://cmake.org/cmake/help/latest/prop_tgt/COMPILE_FLAGS.html).

Also `COMPILE_OPTIONS` is a `;`-separated list which fits `set_property` as opposed to `COMPILE_FLAGS`

Reply:
Yes, it is needed or it needs other modifications:

`shuffle.c` is using `SHUFFLE_USE_AVX2` which is defined as:
 https://github.com/Blosc/c-blosc2/blob/e204359a202ce8624eed47f56bb2093dab43ce60/blosc/shuffle.h#L28-L30
So the definition needs `-mavx2` flag to have `__AVX2__` defined. Same applies for SSE2/NEON/ALTIVEC.

To avoid this, it looks to require to change the conditional include of `[bit]shuffle-*.h` and the way `get_shuffle_implementation` is implemented.

Reply:
I am fine with that.  I just want to make sure we don't abuse of SIMD flags.